### ISSUE TITLE
Using physical data not a "known" function

### ISSUE THREAD

**Issue by will-colea** [SheffieldML/GPyOpt#287]
Hello, I'm new to python and gaussian processes, so I need a little help/insight. I am trying to use GPyOpt with physical x, y data and not a "known" function. The examples all use a known function. I am reading in a .csv file and I want to use bayesian optimization with the test data to let me know what experiment to run next  (obtain next x,y data set). Please any advice and help is appreciated. Thank you.

**Comment by will-colea** [comment]
This is what I am working with now, but I'd like to use x,y collected data instead of the f(x) as some function. Thanks!

# --- Load GPyOpt
from GPyOpt.methods import BayesianOptimization
import numpy as np

# --- Define your problem
def f(x): return (6*x-2)**2*sin(x)
domain = [{'name': 'var_1', 'type': 'continuous', 'domain': (0,10)}]

# --- Solve your problem
myBopt = BayesianOptimization(f=f, domain=domain)
myBopt.run_optimization(max_iter=10)
myBopt.plot_acquisition()

**Comment by apaleyes** [comment]
have a look at external function evaluation example, sounds exactly like what you are doing: https://nbviewer.jupyter.org/github/SheffieldML/GPyOpt/blob/master/manual/GPyOpt_external_objective_evaluation.ipynb

**Comment by will-colea** [comment]
Thank you for referring me to this example. I am trying to work through it currently. I am working on defining my own external evaluation of the function, but I am having difficulties with feeding this into the example now. I am trying to create a function where I can input X[n] and then return the corresponding Y[n] from a .csv file of collected data. Here is what I am working with:

```
# used to be func = GPyOpt.objective_examples.experiments1d.forrester() 
def f(self,X):
        X = X.reshape((len(X),1))
        n = X.shape[0]
        index=list.index(X)
        fval = Y[index]
        return fval # originally fval.reshape(n,1)
```

But when I attempt to run the following section of the example...
```
X_init = df['Y2O3 Sputtering Power (W)']
Y_init = func.f(X_init)

iter_count = 10
current_iter = 0
X_step = X_init
Y_step = Y_init

while current_iter < iter_count:
    bo_step = GPyOpt.methods.BayesianOptimization(f = None, domain = domain, X = X_step, Y = Y_step)
    x_next = bo_step.suggest_next_locations()
    y_next = func.f(x_next)
    
    X_step = np.vstack((X_step, x_next))
    Y_step = np.vstack((Y_step, y_next))
    
    current_iter += 1
```

I get this error...
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-105-21ac2f0d54e7> in <module>
      1 X_init = df['Y2O3 Sputtering Power (W)']
----> 2 Y_init = func.f(X_init)
      3 
      4 iter_count = 10
      5 current_iter = 0

~/opt/anaconda3/lib/python3.7/site-packages/GPyOpt/objective_examples/experiments1d.py in f(self, X)
     35 
     36         def f(self,X):
---> 37                 X = X.reshape((len(X),1))
     38                 n = X.shape[0]
     39                 fval = ((6*X -2)**2)*np.sin(12*X-4)

~/opt/anaconda3/lib/python3.7/site-packages/pandas/core/generic.py in __getattr__(self, name)
   5177             if self._info_axis._can_hold_identifiers_and_holds_name(name):
   5178                 return self[name]
-> 5179             return object.__getattribute__(self, name)
   5180 
   5181     def __setattr__(self, name, value):

AttributeError: 'Series' object has no attribute 'reshape'

I tried looking at the 1d example code that you input into this example, but am still struggling to implement this example properly for my application. Please if you have any other advice, please let me know. Thank you.

**Comment by will-colea** [comment]
Thank you for the recommendation. 
I've taken a look at it. It seems that I need the x input to be an n by 1 array and the externally evaluated function needs to output some n by 1 array. I am still a bit confused how to implement this with xy data from say a .csv file. In this example it still seems like you need a known function. For instance in this 1D objective function, theres a known function defined as...
`fval = 0.25*np.sin(12*X-4)`
How can I implement this without having a known fval? Any thoughts or words of advice?
Thank you for your help and work on this library!

**Comment by apaleyes** [comment]
You don't need to have a known function. On each of the iterations you just need to have a new pair of x,y ready. It could be produced by a function, an experiment, or read of a file. In your case you could ignore what BO recommends in terms of a next x location, and just read next line from the csv. not sure how effective that is, but that's as close as it gets.

on the other hand, since you already have all the data available in the csv, there isn't any benefit in using BO at all. just fit a GP to all this data and find min/max of the final mean function

**Comment by will-colea** [comment]
Thank you for responding so quickly again. Your help is much appreciated. 

Okay, I can definitely make a called function that provides the corresponding y from the data without actually having it plug x into a known function. I will try that. 

Although, this leads me into another question about this for my implementation. To clarify I have a set of about 5 experiments that I used a latin square to choose my initial x points. Each new experiment takes me about a month to complete, which is why I am trying to implement this technique on my data. I want a recommendation on where to sample next in order to find a global max or min for y. In your opinion, do you believe this GPyOpt is viable to accomplish this?

Thank you for your time again!

**Comment by will-colea** [comment]
Thank you again for your response and your time/expertise. 

Yes, so I used a latin square (common in design of experiments) that basically gave me a set of initial sets of experiments to run. I have completed those initial experiments and have this data, but I need to run probably a few more to know if I've reached an optimum. I was hoping to use the results of these initial experiments with bayesian optimization + acquisition function to determine my next experiment to run.
